src/internal/channel.js

```diff
@@ -13,9 +13,21 @@ export function emitter() {
     return () => remove(subscribers, sub)
   }
 
+  function emitAll(items) {
+    for (var i = 0, len = items.length; i < len; i++) {
+      emit(items[i])
+    }
+  }
+
   function emit(item) {
+    if (item === undefined) {
+      return
+    }
+    if (Array.isArray(item)) {
+      emitAll(item);
```

Comment:
should be `return`ed after `emitAll` as it would emit twice in your scenario
1. batch action
1. unpacked batch actions